### PR TITLE
POST /settings

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-1886

### CODE REVIEW

File: api/http/api_useradm.go [hunk 1]
```diff
@@ -404,3 +406,29 @@ func (u *UserAdmApiHandlers) DeleteTokensHandler(w rest.ResponseWriter, r *rest.
...
+
+	var settings map[string]interface{}
+
+	err := r.DecodeJsonPayload(&settings)
+	if err != nil {
+		rest_utils.RestErrWithLog(w, r, l, errors.Wrap(err, "failed to parse request body"), http.StatusBadRequest)
+		return
+	}
+
+	if len(settings) == 0 {
```
Comment (kjaskiewiczz) [inline api/http/api_useradm.go]: it's ok to set empty object, imo

File: api/http/api_useradm.go [hunk 2]
```diff
@@ -404,3 +406,29 @@ func (u *UserAdmApiHandlers) DeleteTokensHandler(w rest.ResponseWriter, r *rest.
...
+func (u *UserAdmApiHandlers) SaveSettingsHandler(w rest.ResponseWriter, r *rest.Request) {
+	ctx := r.Context()
+
+	l := log.FromContext(ctx)
+
+	var settings map[string]interface{}
+
+	err := r.DecodeJsonPayload(&settings)
+	if err != nil {
+		rest_utils.RestErrWithLog(w, r, l, errors.Wrap(err, "failed to parse request body"), http.StatusBadRequest)
```
Comment (kjaskiewiczz) [inline api/http/api_useradm.go]: do we want to return error from json decoder? or something more generic like setting object must be json or something like that?

File: api/http/api_useradm.go (hunk 1 above)
Comment (mchalski) [reply api/http/api_useradm.go]: fixed

File: api/http/api_useradm.go (hunk 2 above)
Comment (mchalski) [reply api/http/api_useradm.go]: yeah, it's enough to say 'cannot parse as json', fixed

**Pull request merged**